yjit/src/backend/ir.rs

```diff
@@ -227,28 +229,26 @@ impl Opnd
         Self::match_num_bits_iter(opnds.iter())
     }
 
-    /// Calculate Opnd::Stack's index from the stack bottom.
-    pub fn stack_idx(&self) -> u8 {
-        self.get_stack_idx().unwrap()
+    /// Convert Opnd::Stack into RegMapping
+    pub fn reg_mapping(&self) -> RegOpnd {
+        self.get_reg_mapping().unwrap()
     }
 
-    /// Calculate Opnd::Stack's index from the stack bottom if it's Opnd::Stack.
-    pub fn get_stack_idx(&self) -> Option<u8> {
-        match self {
-            Opnd::Stack { idx, stack_size, .. } => {
-                Some((*stack_size as isize - *idx as isize - 1) as u8)
-            },
-            _ => None
-        }
-    }
-
-    /// Get the index for stack temp registers.
-    pub fn reg_idx(&self) -> usize {
-        match self {
-            Opnd::Stack { .. } => {
-                self.stack_idx() as usize % get_option!(num_temp_regs)
-            },
-            _ => unreachable!(),
+    /// Convert an operand into RegMapping if it's Opnd::Stack
+    pub fn get_reg_mapping(&self) -> Option<RegOpnd> {
```

Comment:
I think this method should be renamed `get_reg_opnd` to avoid confusion with `RegMapping`.

Reply:
Thanks, that was an oversight. https://github.com/ruby/ruby/pull/11157/commits/e52cca0d3398da1d2200ee7a589f2bb44f2c7b10